docs/customizations/header.md

```diff
@@ -0,0 +1,301 @@
+  - [IHeaderStyleProps](#iheaderstyleprops)
+- [Sample Scenarios](#sample-scenarios)
+  - [Changing header title and icon](#changing-header-title-and-icon)
+  - [Changing button icons](#changing-button-icons)
+  - [Hiding sub-components](#hiding-sub-components)
+  - [Adding a custom button](#adding-a-custom-button)
+  - [Adding a custom image](#adding-a-custom-image)
+  - [Changing element styles](#changing-element-styles)
+
+## Interfaces
+
+### [IHeaderProps](https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/components/header/interfaces/IHeaderProps.ts)
+
+The top-level interface for customizing `Header`.
+
+| Attribute | Type | Required | Description | Default |
+| - | - | - | - | - |
+| componentOverrides     | [IHeaderComponentOverrides](#iheadercomponentoverrides)     | No | Used for overriding default `Header` components, e.g., icon, title, minimize button, and close button | -
+controlProps | [IHeaderControlProps](#iheadercontrolprops) | No | Properties that control the element behariors | -
+styleProps | [IHeaderStyleProps](iheaderstyleprops) | No | Properties that control the element styles | -
+
+### [IHeaderComponentOverrides](https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/components/header/interfaces/IHeaderComponentOverrides.ts)
+
+Custom React components can be passed as input to override the default sub-components. Alternatively, you can stringify the React component before passing it in. The `chat-components` library provides one util function that can be used: [encodeComponentString](https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/common/encodeComponentString.ts).
+
+| Attribute | Type | Required | Description | Default | 
+| - | - | - | - | - |
+| headerIcon     | ReactNode\|string     | No | Used for overriding default header icon | -
+headerTitle | ReactNode\|string | No | Used for overriding default header title | -
+headerMinimizeButton | ReactNode\|string | No | Used for overriding default minimize button | -
+headerCloseButton | ReactNode\|string | No | Used for overriding default close button | -
+
+### [IHeaderControlProps](https://github.com/microsoft/omnichannel-chat-widget/blob/main/chat-components/src/components/header/interfaces/IHeaderControlProps.ts)
+
+| Attribute | Type | Required | Description | Default | 
+| - | - | - | - | - |
+| id     | string     | No | The top-level element id for the header | "oc-lcw-header"
+hideIcon | boolean | No | Whether to hide the icon on the header | false
+hideTitle | boolean | No | Whether to hide the title string on the header | false
+hideMinimizeButton | boolean | No | Whether to hide the minimize button on the header | false
+hideCloseButton | ReactNode\|string | No | Whether to hide the close button on the header | false
+onMinimizeClick | () => void | No | The callback function that will be triggered when the minimize button is clicked | [Minimizes the chat widget]
```

Comment:
default for this and onCloseClick -> https://github.com/microsoft/omnichannel-chat-widget/blob/75135c54d4377cc372876e0a57743df3e86d2bd3/chat-components/src/components/header/common/defaultProps/defaultHeaderControlProps.ts#L11C1-L12C65

Reply:
These are later overwritten in the actual stateful components, so it's not accurate to link that here

Reply:
also same concern as for below comment